src/meshweb/static/admin/map.js

```diff
@@ -162,301 +183,162 @@ async function getNewSelectedNodes(){
-            await updateAdminContent(location.href, {}, false);
-            updateMapForLocation();
-        }
-        handler()
-        // console.log(location.href);
-        event.preventDefault()
-    }, false)
+// Helper function to wrap everything that needs to happen when the admin panel
+// loads
+async function onAdminPanelLoad() {
+    const adminPanelIframeUrl = admin_panel_iframe.contentWindow.location.href;
 
-    // Form submissions
-    window.addEventListener("submit", function (event) {
-        const form = event.target;
-        // Exit early if this navigation shouldn't be intercepted
-        if (shouldNotIntercept(form)) return;
+    // Save the new admin location. We do this here because it means that the admin panel has
+    // recently reloaded.
+    const meshdbLastPageVisited = new URL(adminPanelIframeUrl).pathname;
+    localStorage.setItem(MESHDB_LAST_PAGE_VISITED, meshdbLastPageVisited);
 
-        async function handler() {
-            const formData = new FormData(form);
-            const method = form.method;
+    // Update the URL bar in the browser for viz
+    window.history.pushState("MeshDB Admin Panel", "", meshdbLastPageVisited.replace("/panel", ""));
 
-            if (method.toUpperCase() === "POST") {
-                const submitButton = event.submitter;
-                if (submitButton.getAttribute('name')) {
-                    formData.append(submitButton.getAttribute('name'), submitButton.getAttribute('value'));
-                }
-                await updateAdminContent(form.action, {method: "POST", body: formData});
-            } else if (method.toUpperCase() === "GET") {
-                const params = new URLSearchParams(formData).toString();
-                const actionWithoutParams = new URL(form.action);
-                actionWithoutParams.search = "";
-                await updateAdminContent(`${actionWithoutParams.href}?${params}`);
-            }
+    // Finally, update the map view
+    updateMapLocation(adminPanelIframeUrl);
+}
 
-            updateMapForLocation();
-        }
-        handler()
-        event.preventDefault();
-    })
+// Configures the listener that updates the map based on admin panel activity
+async function listenForAdminPanelLoad() {
+    admin_panel_iframe.addEventListener("load", onAdminPanelLoad);
 }
 
-async function nodeSelectedOnMap(selectedNodes) {
-    if (!selectedNodes) return;
-    if (selectedNodes.indexOf("-") !== -1) return;
+// See above
+async function dontListenForAdminPanelLoad() {
+    admin_panel_iframe.removeEventListener("load", onAdminPanelLoad);
+}
 
-    let selectedNodeInt = parseInt(selectedNodes);
-    if (selectedNodeInt >= 1000000) {
-        /* Hack for APs to not break things. We unfortantely can't do a lot better than this without much pain*/
-        return;
+// Checks local storage for the last page the user navigated to, and directs them
+// there
+async function adminPanelRestoreLastVisited() {
```

Comment:
Do we really want this behavior?

Reply:
Unclear. I used this to make the interface survive a reload without losing the user's place and realized as a side effect the admin interface now picks up where you left off. Could be kind of nifty. Maybe we just leave it and see if people complain?

Reply:
What do you mean "survive a reload"? Doesn't the URL keep track of the page they're on?

Reply:
I don't love keeping state separate from the browser's own state, I feel like that exposes us to potential race conditions that jerk the user to an unexpected place

Reply:
> survive a reload

There was a point when reloading would send you back to the home, because I didn't do the URL stuff yet. Now that I have, we don't need this anymore.